packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart

```diff
@@ -0,0 +1,238 @@
+  _InstanceColumn(this.dataPart)
+      : super(
+          columnTitle(dataPart),
+          fixedWidthPx: scaleByFontFactor(80.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  final _DataPart dataPart;
+
+  static String columnTitle(_DataPart dataPart) {
+    switch (dataPart) {
+      case _DataPart.created:
+        return 'New';
+      case _DataPart.deleted:
+        return 'Deleted';
+      case _DataPart.delta:
+        return 'Delta';
+    }
+  }
+
+  @override
+  int getValue(DiffClassStats classStats) {
+    switch (dataPart) {
+      case _DataPart.created:
+        return classStats.total.created.instanceCount;
+      case _DataPart.deleted:
+        return classStats.total.deleted.instanceCount;
+      case _DataPart.delta:
+        return classStats.total.delta.instanceCount;
+    }
+  }
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class _SizeColumn extends ColumnData<DiffClassStats> {
+  _SizeColumn(this.dataPart, this.sizeType)
+      : super(
+          columnTitle(dataPart),
+          fixedWidthPx: scaleByFontFactor(80.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  final _DataPart dataPart;
+  final _SizeType sizeType;
+
+  static String columnTitle(_DataPart dataPart) {
+    switch (dataPart) {
+      case _DataPart.created:
+        return 'Allocated';
+      case _DataPart.deleted:
+        return 'Freed';
+      case _DataPart.delta:
+        return 'Delta';
+    }
+  }
+
+  @override
+  int getValue(DiffClassStats classStats) {
+    switch (sizeType) {
+      case _SizeType.shallow:
+        switch (dataPart) {
+          case _DataPart.created:
+            return classStats.total.created.shallowSize;
+          case _DataPart.deleted:
+            return classStats.total.deleted.shallowSize;
+          case _DataPart.delta:
+            return classStats.total.delta.shallowSize;
+        }
+      case _SizeType.retained:
+        switch (dataPart) {
+          case _DataPart.created:
+            return classStats.total.created.retainedSize;
+          case _DataPart.deleted:
+            return classStats.total.deleted.retainedSize;
+          case _DataPart.delta:
+            return classStats.total.delta.retainedSize;
+        }
+    }
+  }
+
+  @override
+  String getDisplayValue(DiffClassStats classStats) => prettyPrintBytes(
+        getValue(classStats),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class ClassesTableDiff extends StatefulWidget {
+  const ClassesTableDiff({
+    Key? key,
+    required this.classes,
+    required this.controller,
+  }) : super(key: key);
+
+  final DiffHeapClasses classes;
+  final DiffPaneController controller;
+
+  @override
+  State<ClassesTableDiff> createState() => _ClassesTableDiffState();
+}
+
+class _ClassesTableDiffState extends State<ClassesTableDiff>
```

Comment:
my Table fix PR was just merged, so if you merge with master, you can make this widget stateless, make all the columns / column groups static final, and remove the sorting logic. This will apply for any other tables that are a part of this PR too